### ISSUE TITLE
TSM set point-in-time recovery date via configuration

### ISSUE THREAD

**Issue by schlomo** [rear/rear#1948]
For a special use case it would help if we can set the `TSM_RESTORE_PIT_DATE` and `TSM_RESTORE_PIT_TIME` variables in configuration (or set something else instead) so that I don't have to enter this information on each recovery.

Context is doing a mass recovery from a specific point in time backup.

Currently those variables are actually ignored by https://github.com/rear/rear/blob/aa7b1979c347a3800959d757f7ce485fc10bd788/usr/share/rear/verify/TSM/default/390_request_point_in_time_restore_parameters.sh

**Solution idea**

Remove those variables from `default.conf` as they are actually internal to this script. Instead, switch the user input in the TSM scripts to use the new `UserInput` function written by @jsmeix so that then the user can pre-fill that with the desired answer.

**Comment by schlomo** [comment]
Yes, I would do that if those variables would have worked. Since they never worked I think removing them is fine as it won't break anybody's stuff.

I converted them to lower case in the code and `unset` them at the end to make it clear that these variables are, in fact, internal.

So, unless it is specifically important to you, I'd prefer to leave them out. I think as we convert more and more `read` calls to `UserInput` people will learn to use this new mechanism.

**Comment by gdha** [comment]
@schlomo I'm not a TSM user, so for me it is OK. Trimming the `default.conf` file is not a bad idea. For a novice user it can be overwhelming IMHO.

**Comment by schlomo** [comment]
IMHO we should only have variables in the `default.conf` that are meant for the user to change. By that logic those variables don't belong there.

Once upon a time we said that `default.conf` should initialize *every* variable, but this example shows that it is better if it contains only variables meant for users but not internal variables.

**Comment by jsmeix** [comment]
What about implementing it this way?
https://github.com/rear/rear/pull/1951

**Comment by jsmeix** [comment]
In current ReaR master code
https://raw.githubusercontent.com/rear/rear/master/usr/share/rear/conf/default.conf
reads
<pre>
# Here we define and describe all configuration variables and set them to a default.
</pre>
so that 'default.conf' is menat to initialize only all those variables
that are meant for users to be changed (if needed).

Accordingly it is right to remove variables from default.conf
that are not meant as configuration variables for users, cf.
all those non-config variables that are set in usr/sbin/rear

@schlomo 
your
https://github.com/rear/rear/commit/f5558efbf5407437bc3c6e671146faefa9992ee4
looks good to me from plain looking at the code
(I cannot test proprietary backup tools).

The only tiny inconsistency is the date format that the code does
<pre>
tsm_restore_pit_date=$( date -d "$answer" +%Y.%m.%d 2>/dev/null ) ...
</pre>
versus what date format is told the user in the LogPrint message
<pre>
LogPrint " ... ${tsm_restore_pit_date} ${tsm_restore_pit_time} (MM/DD/YYYY HH:mm:ss)"
</pre>
because (as far as I understand the copd from plain looking at it)
${tsm_restore_pit_date} is e.g. '2018.11.08' and not '11/08/2018', cf.
https://github.com/rear/rear/pull/1951#issuecomment-435874716

FYI
some explanations about my ideas behind my
https://github.com/rear/rear/pull/1951

I thought TSM_RESTORE_PIT_DATE and TSM_RESTORE_PIT_TIME
are meant as user config variables (because they are in default.conf)
but implementing the code for that had been somehow forgotten.

Because 390_request_point_in_time_restore_parameters.sh
runs during "rear recover" I found it a bit too hard to error out
only because the user may have by accident a typo in his input
so that I tried to implement an endless loop until input was valid.
But because 390_request_point_in_time_restore_parameters.sh
runs early during "rear recover" (the verify stage runs at the beginning
in particular before the disklayout gets recreated) I think it does
not hurt in practice to error out at that early stage of "rear recover".

**Comment by schlomo** [comment]
@skrueger8 FYI
